### PR TITLE
Query crossing entity boundaries

### PR DESCRIPTION
**Pull request by bserdar**

### CODE REVIEW

File: crud/src/main/java/com/redhat/lightblue/mediator/QueryPlanNodeExecutor.java [hunk 1]
```diff
@@ -224,23 +267,23 @@ private void execute(OperationContext ctx,
...
-            if(parents!=null) {
-                for(ChildDocReference parent:parents) {
-                    resultDoc.setParentDoc(parent.getDocument().getQueryPlanNode(),parent);
-                    parent.getChildren().add(resultDoc);
-                    LOGGER.debug("Linked parent ref:{}",parent);
+
+            if(parents!=null&&!parents.isEmpty()) {
+                for(DocReference parent:parents) {
+                    if(parent instanceof ChildDocReference) {
+                        QueryPlanNode parentNode=parent.getDocument().getQueryPlanNode();
```
Comment (jewzaam) [inline crud/src/main/java/com/redhat/lightblue/mediator/QueryPlanNodeExecutor.java]: Line "QueryPlanNode parentNode=parent.getDocument().getQueryPlanNode();" can be removed.

File: crud/src/main/java/com/redhat/lightblue/assoc/ResultDoc.java [hunk 2]
```diff
@@ -132,64 +139,103 @@ public void setParentDoc(QueryPlanNode parentNode,ChildDocReference ref) {
...
+     */
+    public static void associateDocs(List<ResultDoc> parentDocs,
+                                     List<ResultDoc> childDocs,
+                                     List<Conjunct> edgeClauses,
+                                     CompositeMetadata root) {
+        // We have to find a list of childDocs for every reference in a parent doc
+        LOGGER.debug("Associating {} parent docs",parentDocs.size());
+        if(!childDocs.isEmpty()&&!parentDocs.isEmpty()) {
+            QueryPlanNode childDocsNode=childDocs.get(0).node;
+            QueryPlanNode parentDocsNode=parentDocs.get(0).node;
```
Comment (jewzaam) [inline crud/src/main/java/com/redhat/lightblue/assoc/ResultDoc.java]: Is it ever the case with multiple parent or child docs that the query plan node is different for different elements in those lists?  The bindingInfo is created with the plan node from the 0th element.

File: crud/src/main/java/com/redhat/lightblue/mediator/QueryPlanNodeExecutor.java (hunk 1 above)
Comment (bserdar) [inline crud/src/main/java/com/redhat/lightblue/mediator/QueryPlanNodeExecutor.java]: Removed.

File: crud/src/main/java/com/redhat/lightblue/assoc/ResultDoc.java (hunk 2 above)
Comment (bserdar) [inline crud/src/main/java/com/redhat/lightblue/assoc/ResultDoc.java]: Those lists are documents associated with a query plan node, meaning, each
list contains documents of one entity type only.  Added some more comments.

**Pull request merged**